src/ensembl/src/content/app/entity-viewer/gene-view/components/default-transcripts-list/DefaultTranscriptsList.tsx

```diff
@@ -29,12 +33,18 @@ import styles from './DefaultTranscriptsList.scss';
 type Props = {
   gene: Gene;
   rulerTicks: TicksAndScale;
+  push: Push;
 };
 
 const DefaultTranscriptslist = (props: Props) => {
+  const params: { [key: string]: string } = useParams();
   const { gene } = props;
   const sortedTranscripts = defaultSort(gene.transcripts);
 
+  useEffect(() => {
+    props.push(getGeneViewPath(params, null));
+  }, []);
```

Comment:
Sorry, I'm confused.
1) Why are you doing this in multiple components? Surely it should be enough to just have one useEffect responsible for reconciling redux state with the url
2) What are the situations when the url has to be reconciled with the redux state? I thought it's only when the species changes.
3) I have a suspicion the current code not work when we start switching between species. It's working now; but the reason may be that every time we switch between species we are unmounting and re-mounting components.

Reply:
> What does this code do? You are getting params from the router — and then you update the router with the same params? Why?

How else can we generate the URL that includes those params using `urlHelper`?

> Why does the reconciliation code exist in multiple components? Surely it should be sufficient to just have one useEffect responsible for reconciling redux state with the url

This is because we aren't using `Link` for the tabs and instead using the `onTabChange` function, which we pass onto `Tabs`, make the changes.

> What are the situations when the url has to be reconciled with the redux state? I thought it's only when the species changes.

We also need to update redux when the view mode is changed e.g. protein -> variation

> I have a suspicion the current code not work when we start switching between species. It's working now; but the reason may be that every time we switch between species we are unmounting and re-mounting components.

I have not looked into this scenario properly and should definitely check it.

Reply:
> This is because we aren't using Link for the tabs and instead using the onTabChange function, which we pass onto Tabs, make the changes.

I am lost. What you are saying does not explain what is going on in the `useEffect` hook. The hook doesn't call `onTabChange`.

Reply:
> I am lost. What you are saying does not explain what is going on in the useEffect hook. The hook doesn't call onTabChange.

Sorry for the confusion. I actually need to update the code to reflect this in GeneRelationships and GeneFunction. I'll point it out once I've pushed the code.

Reply:
@azangru I have changed the code to make the `onTabChange` function to be reused in `useEffect` for GeneRelationships and GeneFunction.

The DefaultTranscriptsList functions in a different way as it it doesn't have any tabs. However, the view param needs to change when navigated to it from a different view mode such as protein or variation.

Reply:
> Wouldn't it be better if the url replacement is done where tab change is triggered?

It is a tricky one as only the transcript tab of GeneViewTabs has this behaviour. The other two (GeneRelationships and GeneFunction) have child tabs that change the URL.